src/components/Controls.vue

```diff
@@ -3,13 +3,23 @@ import Vue from 'vue';
 import Legend from '@/components/MultiLink/Legend.vue';
 
 import store from '@/store';
-import { Node, Link } from '@/types';
+import { Node, Link, Network } from '@/types';
 
 export default Vue.extend({
   components: {
     Legend,
   },
 
+  data(): {
+    searchTerm: string;
+    searchErrors: string[];
+    } {
+    return {
+      searchTerm: '',
+      searchErrors: [],
+    };
+  },
```

Comment:
Specifically for `data()` functions in Vue components, I like the following pattern a lot:

```suggestion
  data(): {
    return {
      searchTerm: '' as string,
      searchErrors: [] as string[],
    };
  },
```

This is one of the few times I use `as` without worrying about it.